src/test/java/org/folio/rest/api/InstanceStorageTest.java

```diff
@@ -1881,13 +1892,38 @@ public void canPostSynchronousBatch() throws Exception {
     JsonObject uprooted         = instancesArray.getJsonObject(0);
     JsonObject smallAngryPlanet = instancesArray.getJsonObject(numberOfInstances / 2);
     JsonObject temeraire        = instancesArray.getJsonObject(numberOfInstances - 1);
+
     assertExists(uprooted);
     assertExists(smallAngryPlanet);
     assertExists(temeraire);
 
     assertNotSuppressedFromDiscovery(instancesArray);
   }
 
+  @Test
+  public void instancesCreatedInABatchShouldHaveStatusDate() throws Exception {
+    JsonArray instancesArray = new JsonArray();
+    
+    for (int i = 0; i < 3; i++) {
+      instancesArray.add(smallAngryPlanet(UUID.randomUUID()));
+    }
+
+    JsonObject instanceCollection = new JsonObject().put(INSTANCES_KEY, instancesArray);
+
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+    client.post(instancesStorageSyncUrl(""), instanceCollection, TENANT_ID, ResponseHandler.empty(createCompleted));
+    assertThat(createCompleted.get(30, SECONDS), statusCodeIs(HttpStatus.HTTP_CREATED));
+
+    Response resp;
+    for (int i = 0; i < 3; i++) {
+      resp = getById(instancesArray.getJsonObject(i).getString("id"));
+      assertThat(resp.getJson().getString(STATUS_UPDATED_DATE_PROPERTY), notNullValue());
+      assertThat(resp.getJson().getString(STATUS_UPDATED_DATE_PROPERTY), hasIsoFormat());
+    }
```

Comment:
Similarly to my feedback on the other pull request, I think this can be changed to use the `JSONArrayHelper` and `forEach`.